content-services/content-sources-backend
# Refs 3125: allow delete repo during introspect/snapshot

## Summary
This changes the behavior of the delete and bulk delelete repository API endpoints to allow deletion if a snapshot is in progress. If a snapshot is in progress, the snapshot task is canceled. The introspect task is also canceled.

The corresponding frontend PR is here: https://github.com/content-services/content-sources-frontend/pull/269

## Testing steps
1. Create or bulk create repositories. It's easiest to do this in the UI.
2. While the status is pending, delete the repositories.
3. You should find that the running or queued (introspect or snapshot) tasks are canceled, and the repositories are deleted.

## Checklist
